src/Language/ASKEE/Exposure/Interpreter.hs

```diff
@@ -124,10 +130,12 @@ suspend v
 data EvalRead = EvalRead
   { erPrecomputed :: Map Value Value
   , erLocalVars   :: Map Ident Value
+  , erPutFileFn   :: FilePath -> LBS.ByteString -> IO ()
+  , erGetFileFn   :: FilePath -> IO LBS.ByteString
   }
 
 emptyEvalRead :: EvalRead
-emptyEvalRead = EvalRead Map.empty Map.empty
+emptyEvalRead = EvalRead Map.empty Map.empty LBS.writeFile LBS.readFile
```

Comment:
Now that `EvalRead` contains more than just `Map`s, I wonder if the name `emptyEvalRead` is still a suitable name for this function. Really, this choice of `erPutFileFn` and `erGetFileFn` are tailor-made for an application like `champ`, which doesn't concern itself with server-client interactions like `donu` does.

I'd advocate for:

1. Replacing `emptyEvalRead` with a function that takes arguments of the appropriate types for `erPutFileFn` and `erGetFileFn`,
2. Make `evalLoop` take an `EvalRead` as an argument (which would obviate the need for `evalLoop'`), and
3. Force each consumer (currently, `champ` and `donu`) to make a choice of what `erPutFileFn` and `erGetFileFn` should be when constructing the `EvalRead` to pass to `evalLoop`.

This is slightly more verbose, but I think it separates concerns more cleanly.

Reply:
Yeah this seems reasonable